src/lambda_function/s3_event_config/app.py

```diff
@@ -56,35 +161,117 @@ def add_notification(
         bucket (str): bucket name of the s3 bucket to add the config to
         bucket_key_prefix (str): bucket key prefix for where to look for s3 object notifications
     """
-    s3_client.put_bucket_notification_configuration(
-        Bucket=bucket,
-        NotificationConfiguration={
-            f"{destination_type}Configurations": [
-                {
-                    f"{destination_type}Arn": destination_arn,
-                    "Events": ["s3:ObjectCreated:*"],
-                    "Filter": {
-                        "Key": {
-                            "FilterRules": [
-                                {"Name": "prefix", "Value": f"{bucket_key_prefix}/"}
-                            ]
-                        }
-                    },
-                }
-            ]
+    update_required = False
+    destination_type_arn = f"{destination_type}Arn"
+    new_notification_configuration = {
+        destination_type_arn: destination_arn,
+        "Events": ["s3:ObjectCreated:*"],
+        "Filter": {
+            "Key": {
+                "FilterRules": [{"Name": "prefix", "Value": f"{bucket_key_prefix}/"}]
+            }
         },
+    }
+
+    (
+        existing_bucket_notification_configuration,
+        existing_notification_configurations_for_type,
+    ) = get_existing_bucket_notification_configuration_and_type(
+        s3_client, bucket, destination_type
     )
-    logger.info("Put request completed....")
 
+    (
+        index_of_matching_arn,
+        matching_notification_configuration,
+    ) = get_matching_notification_configuration(
+        destination_type_arn,
+        existing_notification_configurations_for_type,
+        destination_arn,
+    )
 
-def delete_notification(s3_client: boto3.client, bucket: str):
-    """Deletes the S3 notification configuration from an existing bucket
+    if index_of_matching_arn is not None:
+        if json.dumps(
+            matching_notification_configuration, sort_keys=True
+        ) != json.dumps(new_notification_configuration, sort_keys=True):
```

Comment:
I still think it would be cleaner to compare the "index" of these two notifications configurations, the index being the necessary properties `Events` and `Filter.Key.FilterRules.Value` in our notification configuration. I think it's less likely that these fields will change in the future versus some arbitrary change in the response. But this works for now 👍

Reply:
I like this thought. I will extract this out to a `compare` function so all the fields we are checking is in one spot.

Reply:
I added this to: https://github.com/Sage-Bionetworks/recover/pull/87/commits/b473c56fcd1a4c4d35131fcf366c740e0564fee7